pages/Documents/MessagingChannels/ConnectToMessaging/api.md

```diff
@@ -0,0 +1,254 @@
+
+- When using app JWT to call C2M API, a response below indicates the jwt is expired and new app jwt to be obtained from sentinel api.
+	```json
+	{
+		"code": 0,
+		"message": "jwt expired"
+	}
+	```
+
+<strong>Do we need any other JWT other than APP JWT e.g. Consumer JWT?</strong>
+C2M service does not create or consume consumer JWT or other JWT except APP JWT. C2M API consumes AppJWT created from provided clientId and Secret for authentication.
+
+<strong>What should the authentication header look like, is the bearer token the only thing required even in production usage? Do we need to include our ConsumerKey/Secret or our AccessToken/Secret that we use in the 1.0 API at all, or any other information?</strong>
+App Jwt will be consumed as Bearer Token. No other key, secret or token will be consumed by C2M Messaging api.
+
+<strong>How does C2M 2.0 api provide status of message e.g. success/failure of delivery ?</strong>
+The C2M 2.0 campaign api is asynchronous meaning that the success and failure of a message to a recipient is noted only when the recipient is picked from the C2M internal queue and a message is sent as per pre configured message rate. The [conversations](#conversations-api-example-request-and-response) api will provide the status of recipients tied to the campaign created.
+
+<strong>Of the error cases described above, which of those errors should we consider "retry-able"? For example, a bad request due to a missing field is not retry-able because it will just always fail, but a case where one of the downstream services was temporarily unavailable could warrant a retry. Which error cases that we could get back from the /campaigns endpoint are retryable and how should we handle a retry to avoid sending duplicate messages to a customer?</strong>
+C2M Messaging service has retry mechanism internally on dependent services to reduce failures due to transient errors.
+
+<strong>What’s the lookback period?</strong>
+- Lookback period is how long will LP services maintain context (like campaign info, skill etc) for a reply of a message that is sent to the recipient/consumer.  
+- Lookback period can be pre-configured up to 30 days. Current maximum lookback period is 30 days from when messages are sent using C2M API. Example: When a message is sent to consumer using C2M API and if consumer replies within 30 days from when message was sent, the response will be redirected to LE agent according to specified skill. A response after 30 days will not be treated as a conversation. Please note, if a consumer has an existing active conversation with a brand in any channel, the outbound message won’t be delivered.
+
+<strong>How do we know which field is optional or required?</strong>
+Refer to this document and [swagger](https://app.swaggerhub.com/apis-docs/nsavla/Connect_To_Messaging/1.0.0#/).
+
+
+| Field Name | Limitation |
```

Comment:
what is this table for?

Reply:
This one is the table header for the below.

Reply:
yes, but I do not see a line to talk about what the table is about. and I do not think it is related to the question above
`How do we know which field is optional or required`

Reply:
I'll add seperate question for this.